docs/change_resource_hierarchy.md

```diff
@@ -0,0 +1,350 @@
+        tf_validate "$env_path" "$env" "$policysource" "$component"
+      else
+        #🟢 Replace dash (-) for slash (/) in component to fix warning message
+        echo "$(echo ${component} | sed -r 's/-/\//g' )/$env doesn't match $environments_regex; skipping"
+      fi
+
+      component=$old_component #🟢 Assign old component value before next while-loop iteration
+    done
+    ```
+
+1. Do the same changes in `single_action_runner` function.
+
+    ```text
+    single_action_runner() {
+    local env
+    ...
+    #🟢 Set maxdepth in find command -------------- 🟢
+    find "$component_path" -mindepth 1 -maxdepth $maxdepth -type d | sort -r | while read -r env_path ; do
+        env="$(basename "$env_path")"
+
+        old_component=$component #🟢 Holds component value before call check_env_path_folder
+
+        #🟢 Calls check_env_path_folder to get fixed component value
+        check_env_path_folder "$env_path" "$base_dir" "$component" "$env"
+
+        component=$current_component #🟢 Get fixed component value
+    ...
+    ```
+
+1. Fix warning message for doesn't match directories.
+
+    ```text
+        esac
+      else
+        #🟢 Replace dash (-) for slash (/) in component to fix warning message
+        echo "$(echo ${component} | sed -r 's/-/\//g' )/${env} doesn't match ${branch}; skipping"
+      fi
+
+      component=$old_component #🟢 Assign old component value before next while-loop iteration
+    done
+    ```
+
+1. Create a new function `check_env_path_folder`.
+
+    ```text
+    ...
+    #🟢 New check_env_path_folder function
+
+    ## Fix component name to be different for each environment. It is used as tf-plan file name
+    check_env_path_folder() {
+    local lenv_path=$1
+    local lbase_dir=$2
+    local lcomponent=$3
+    local lenv=$4
+
+    if [[ "$lenv_path" =~ ^($lbase_dir)/(.+)/$lenv ]] ; then
+        # The ${BASH_REMATCH[2]} means the second group in regex expression
+        # This group is the folders between base dir and env
+        # This value guarantees that tf-plan file name will be unique for each environment
+        current_component=$(echo ${BASH_REMATCH[2]} | sed -r 's/\//-/g')
+    else
+        current_component=$lcomponent
+    fi
+    }
+
+    #🟢 End of New check_env_path_folder function
+
+    case "$action" in
+    init|plan|apply|show|validate )
+    ...
+    ```
+
+## Code Changes - Terraform Files
+
+<pre>
+example-organization/
+├── bootstrap
+├── common
+├── <b>development *</b>
+│   ├── finance
+│   └── retail
+├── <b>non-production *</b>
+│   ├── finance
+│   └── retail
+└── <b>production *</b>
+    ├── finance
+    └── retail
+</pre>
+
+*Example 1 - An example of Terraform Foundation Example with hierarchy changed*
+
+### Step 2-environments
+
+1. Create the folder hierarchy for the business units in each environment.
+
+    Example:
+
+    2-environments/envs/development/main.tf
+
+    ```text
+    module "env" {
+        source = "../../modules/env_baseline"
```

Comment:
Couldnt these changes be made within baseline so they are replicated for all envs without any code change?

Reply:
Done